subsys/net/ip/net_context.c

```diff
@@ -2426,6 +2434,22 @@ int net_context_recv(struct net_context *context,
 	return 0;
 }
 
+int net_context_update_recv_wnd(struct net_context *context,
+				int delta)
+{
+#if defined(CONFIG_NET_TCP)
+	if (!context->tcp) {
+		NET_ERR("context->tcp == NULL");
+		return -EPROTOTYPE;
+	}
+
+	context->tcp->recv_wnd += delta;
```

Comment:
> recv_wnd is uint16_t (actually it should be u16_t

Sorry, this is all is based on the older patch made before u*_t refactor. Now fixed.

Ok, to do bounds checking right, we should consider: is u16_t the right type for recv_wnd value (as stored in struct net_tcp)?